$NetBSD: patch-ab,v 1.8 2005/11/21 09:47:45 rillig Exp $

IRIX has initstate and random: do not map them to srand and rand.

--- config.h.orig
+++ config.h
@@ -40,7 +40,7 @@
 #define SYSV /* Linux is SysV */
 #endif
 
-#ifdef SVR3
+#if defined(SVR3) || defined(__sgi)
 #ifndef SYSV
 /* Stupid SGI machines define SVR3 but not SYSV */
 #define SYSV
@@ -65,6 +65,12 @@
 #endif
 #endif
 
+/* agc - test for 4.4 BSD-derived systems properly */
+#include <sys/param.h>
+#if (defined(BSD) && BSD >= 199306)
+#define __BSD_4_4__
+#endif
+
 /* NOTE:  The RESET_PROGRAM resets the terminal to a "normal" state 
    If you comment out the definition, all will be well except that metamail's
    -R switch won't work, and metamail-called programs might be more likely
@@ -100,14 +106,20 @@
 #endif
 
 #ifdef SYSV
+#if !(defined(__sun__) && defined(__svr4__)) && !defined(LINUX) && !(defined(__sun) && defined(__SVR4))
 #define killpg(a, b) kill(-(a), (b))
+#endif
 #define bcopy(a, b, c) memcpy(b, a, c)
+#ifndef LINUX
 #define bzero(a, b) memset(a, 0, b)
+#endif
 #define bcmp memcmp
 #define index strchr
 #define rindex strrchr
+#ifndef __sgi	/* IRIX has initstate and random */
 #define initstate srand
 #define random rand
+#endif
 #define NO_RLIMITS 1
 #define sigtype void
 #endif
@@ -155,9 +167,13 @@
 #ifdef NeXT
 #define sigtype void
 #else
+#if defined(__BSD_4_4__)
+#define sigtype void
+#else
 #define sigtype int
 #endif
 #endif
+#endif
 
 #ifdef MSDOS
 #define PATH_SEPARATOR ';'
@@ -173,7 +189,7 @@
 #else
 #define PATH_SEPARATOR ':'
 #ifndef STDPATH
-#define STDPATH "/.mailcap:/usr/local/etc/mailcap:/usr/etc/mailcap:/etc/mailcap:/etc/mail/mailcap:/usr/public/lib/mailcap"
+#define STDPATH "/.mailcap:" PKG_SYSCONFDIR "/mailcap"
 #endif
 #endif
 #endif
